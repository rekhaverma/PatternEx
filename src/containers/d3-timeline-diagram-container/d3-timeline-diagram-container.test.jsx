import React from 'react';
import moment from 'moment';

import TimelineDiagramContainer from './d3-timeline-diagram-container.component';

jest.mock('./constants', () => ({
  defaultEndDate: { format: () => '2018-08-01' },
  defaultStartDate: { format: () => '2018-08-01' },
}));

/**
 * @status: Complete
 * @sign-off-by: Alex Andries
 */
describe('<TimelineDiagramContainer />', () => {
  it('should match with snapshot', () => {
    const props = {
      startDate: moment.utc('2018-08-06').startOf('day'),
      endDate: moment.utc('2018-08-10').endOf('day'),
      tab: 'malicious',
      items: {},
      description: 'description',
      activeView: 'malicious',
      onDayClick: jest.fn(),
      resetDayClick: jest.fn(),
    };
    const wrapper = shallow(<TimelineDiagramContainer {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
