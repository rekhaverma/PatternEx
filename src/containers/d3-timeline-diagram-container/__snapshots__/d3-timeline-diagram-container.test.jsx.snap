// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelineDiagramContainer /> should match with snapshot 1`] = `
<TimelineDiagram
  activeView="malicious"
  config={
    Object {
      "marginBottom": 5,
      "marginLeft": 20,
      "marginTop": 20,
      "svgHeight": 160,
      "svgWidth": 910,
    }
  }
  endDate={"2018-08-10T00:00:00.000Z"}
  items={Object {}}
  onDayClick={[Function]}
  resetDayClick={[Function]}
  setTimeRange={[Function]}
  startDate={"2018-08-06T00:00:00.000Z"}
  tab="malicious"
  timeRangeEnd={
    Object {
      "format": [Function],
    }
  }
  timeRangeEndPosition={870}
  timeRangeStart={
    Object {
      "format": [Function],
    }
  }
  timeRangeStartPosition={0}
  updateRangePercent={[Function]}
/>
`;
